src/main.ts

```diff
@@ -41,49 +39,32 @@ async function main() {
     client.once(Events.ClientReady, async (c) => {
         console.log(`Ready! Logged in as ${c.user.tag}`);
 
-        // OK, here we loop through guilds and build a service instance for each
-        const GuildIdList = client.guilds.cache.map((guild) => guild.id);
+        // set up instances of services for each server crabapple is one
+        for (const guild of client.guilds.cache.values()) {
+            const config = ConfigInstanceByGuildIdGet(guild.id);
 
-        for (const guildId of GuildIdList) {
-            const instanceId = await ConfigInstanceIdByGuildIdGet(guildId);
-            if (!instanceId) {
-                console.log(`No Instance ID found for guild ${guildId}`);
-            } else {
-                const coreConfig = await ConfigCoreGet();
-                const instanceConfig = await ConfigInstanceByGuildIdGet(
-                    instanceId
-                );
-
-                if (!instanceConfig) {
-                    console.error(
-                        `No config for: ${instanceId} (${guildId})! Guild is not enabled.`
-                    );
-                    // in this case we do not create a serviceMap for the guild
-                    continue;
-                }
-
-                const orgCoreService = new NmOrgService(
-                    coreConfig.GSPREAD_CORE_ORG_ID
-                );
-
-                GuildServiceMap[guildId] = {
-                    foodCountDataInstanceService: new NmFoodCountDataService(
-                        instanceConfig.GSPREAD_FOODCOUNT_ID
-                    ),
-                    foodCountInputInstanceService: new NmFoodCountInputService(
-                        orgCoreService
-                    ),
-                    orgCoreService,
-                    personCoreService: new NmPersonService(
-                        coreConfig.GSPREAD_CORE_PERSON_ID
-                    )
-                };
+            if (!config) {
+                return console.log(`No Instance ID found for guild ${guild.name}`);
```

Comment:
you want a "continue" not a "return" here, otherwise you break any guild instances that come after the missing one.

Reply:
I think you're looking at something outdated. I fixed this yesterday